app/pages/lab-pages-editor/DataManager.jsx

```diff
@@ -0,0 +1,82 @@
+/*
+Data Manager
+Responsible for fetching, updating, and saving Workflow resources from/to
+the Panoptes service.
+ */
+
+import apiClient from 'panoptes-client/lib/api-client'
+import { useEffect, useState } from 'react'
+import { WorkflowContext } from './context.jsx'
+
+export default function DataManager ({
+  // key: to ensure DataManager renders FRESH (with states reset) whenever workflowId changes, use <DataManager key={workflowId} ... />
+  children,
+  workflowId,
+}) {
+  let initialised = false
+  const [ apiData, setApiData ] = useState({
+    workflow: null,
+    status: 'ready',
+  })
+
+  // Fetch workflow when the component loads for the first time.
+  // See notes about 'key' prop, to ensure states are reset:
+  // https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes
+  // Also see general pattern notes:
+  // https://react.dev/reference/react/useEffect#fetching-data-with-effects
+  useEffect(async function whenComponentMounts_doInitialise () {
+    console.log('+++ DataManager: fetchWorkflow', workflowId)
+    if (initialised) return 
+    if (!workflowId) return
```

Comment:
You should return your cleanup function here, as well as later on. Having different return values from the effect, depending on external variables, might lead to behaviour that's difficult to debug later on.

Reply:
Thanks, I missed this major error - I didn't consider that the cleanup function wouldn't be returned in certain cases and would mess things up. 👌 I've completely rewritten the effect.